plugins/ngrok/api_credentials.go

```diff
@@ -0,0 +1,86 @@
+package ngrok
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+	"gopkg.in/yaml.v3"
+)
+
+func APICredentials() schema.CredentialType {
+	return schema.CredentialType{
+		Name:    credname.APIKey,
+		DocsURL: sdk.URL("https://ngrok.com/docs/ngrok-agent/config"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "API Key used to authenticate to ngrok API.",
+				Optional:            true,
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 48,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.TempFile(ngrokApiConfig, provision.Filename("config.yml"), provision.AddArgs("--config", "{{ .Path }}")),
```

Comment:
I recently noticed that ngrok now supports env var for the api credential : https://ngrok.com/docs/ngrok-agent/#supported-env-variables. It could be more straightforward to use that here and get rid of hardcoding `Version: 2` in the file

Reply:
Actually, I opened an issue for this, and we can treat it with another occasion: https://github.com/1Password/shell-plugins/issues/216